[FEAT] 메인 화면 (투데이 탭) UI

hogang1223 opened the pull request:
<img width="200" height="405" alt="스크린샷 2025-08-04 18 44 51" src="https://github.com/user-attachments/assets/0a911926-7ede-4269-9566-37180e24d883" />

### 메인화면 UI
앱의 메인화면인 오늘 하루 요약(투데이) UI 구현했습니다.

1. section별로 뷰 파일을 분리했습니다. (오늘하루, 명언, 추천콘텐츠, 최근기록)
2. 뷰파일을 분리했으나, padding, spacing 값등의 제약사항은 공통으로 사용하기 위해 네임스페이싱 + enum으로 선언했습니다.
``` swift
enum TodaySummary {
    enum Metric {
        static let contentSpacing = 16.0
        static let contentPadding = 20.0
    }
}
```
3. 각 섹션 별 일관된 rounding 처리와 유지보수성, 재사용성를 위해 `RoundedContainer` Common View를 생성했습니다.


**특이사항**
1. SwiftUI 공부를 위한 프로젝트로, SwiftUI로 구현했습니다.
5. 하단에 탭바를 추가할 예정이며, 그 중 첫번째 탭의 화면 입니다 (하단 탭은 추후 개발)
6. 아직 버튼에 action은 추가하지 않았습니다. 추후 비즈니스 로직 연결  혹은 이동할 화면 구현 시 추가할 예정입니다.
7. 오늘의 명언은 API로 데이터를 연결할 예정으로 현재는 임의의 텍스트를 넣어 놓았습니다.

hogang1223 committed:
feat - 메인뷰 (투데이 탭) UI

jeohong commented:
REVIew

jeohong commented on how-was-today/how-was-today/Views/TodaySummaryViews/RecomendationSection.swift:
```diff
@@ -0,0 +1,63 @@
+        RoundedContainer(cornerRadius: 8.0) {
+            HStack(spacing: 8.0) {
+                Image(systemName: systemImageName)
+                    .font(.system(size: 20.0))
```
font 도 일관성을 위해 Color 처럼 Extension으로 빼서 사용하면 어떨까요? 👀

hogang1223 replied:
이게 이미지 사이즈라.. 이번 뷰에서만 이렇게 사용하게되어 따로 Extension으로 정의하지는 않았습니다.
추후 system Font를 더 자주 사용하게 될것으로 보여지면 그 때 Extension으로 정의할까 합니다.
일단은 Metric에 imageSize로 따로 정의해서 매직넘버는 사용하지 않게 했습니다.  🙏

jeohong commented on how-was-today/how-was-today/Views/TodaySummaryViews/RecentRecordSection.swift:
```diff
@@ -0,0 +1,141 @@
+            
+            HStack(spacing: 12) {
+                Circle()
+                    .fill(Color(hex: "#A3DB91"))
```
Color Extension 으로 컬러값을 관리하여 추후에 동일 색상을 사용하게 될 경우를 대비해도 좋을것 같아요 !

hogang1223 replied:
✅ 수정했습니다!

jeohong reviewed (commented):
PR 의 브랜치가 Main 인데 별도의 브랜치 없이 바로 Main에 Merge 하시나요? 👀

이 외의 코드는 너무 간결하고 좋은것 같아요!

Commit:
refact - 최근 기록 섹션 뷰 내부 유사한 디자인 반복으로 재사용할 수 있도록 RecentRecordCardView 추가

- 주석 추가
- hexCode 매직넘버로 정의한 Color extension에 추가
- 각 화면 별 레이아웃 제약사항 매직넘버 > private Metric으로 변경

Commit:
style - TodaySummaryViews 폴더 제거

hogang1223 commented:
지적해주신 merge branch develop으로 변경했습니다 😱
전반적으로 말씀 주신 사항들 수정했습니다.!

- 주석 누락한 각 파일에 주석 추가했습니다.
- Views내에 있던 TodaySummary폴더는 기능별로 나눌 시 기능 폴더 안에 view,model 등의 폴더가 존재하는게 더 나은 것 같아 삭제했습니다.
- 최근 기록 섹션의 경우 내부 유사한 디자인이 반복되고있어 재사용 할 수 있도록 `RecentRecordCardView` 를 추가했습니다.

jeohong reviewed (approved):
LGTM 🥳
